File: src/main/java/demo/kotlin/DatabaseInitializer.java
package demo.kotlin;

import demo.kotlin.model.Cow;
import demo.kotlin.repository.CowRepository;
import org.springframework.boot.CommandLineRunner;
import org.springframework.stereotype.Component;

import java.time.LocalDateTime;

import static java.util.Arrays.asList;
import static reactor.core.publisher.Flux.fromIterable;

@Component
public class DatabaseInitializer implements CommandLineRunner {
    private CowRepository cowRepository;

    public DatabaseInitializer(CowRepository cowRepository) {
        this.cowRepository = cowRepository;
    }

    @Override
    public void run(String... args) throws Exception {
        Cow marguerite = new Cow("Marguerite", LocalDateTime.of(2017, 9, 28, 13, 30));
        Cow laNoiraude = new Cow("La Noiraude");

        // uncomment if targetting a real MongoDB Database (not embedded)
//        cowRepository.deleteAll()
//                .block();

        fromIterable(asList(marguerite, laNoiraude))
                .flatMap( cow -> {
                    System.out.println("saving " + cow.getName());
                    return cowRepository.save(cow);
                }).blockLast();
    }
}

File: src/main/kotlin/demo/kotlin/DatabaseInitializer.kt
package demo.kotlin

import demo.kotlin.repositories.CowRepository
import demo.kotlin.repositories.RoleRepository
import demo.kotlin.repositories.UserRepository
import demo.kotlin.repositories.UserRoleRepository
import org.springframework.boot.CommandLineRunner
import org.springframework.stereotype.Component

@Component
class DatabaseInitializer(
        private val cowRepository: CowRepository,
        private val userRepository: UserRepository,
        private val roleRepository: RoleRepository,
        private val userRoleRepository: UserRoleRepository
) : CommandLineRunner {

    override fun run(vararg args: String) {
        println("DatabaseInitializer start")

        cowRepository.init()
                .then(userRepository.init())
                .then(roleRepository.init())
                .then(userRoleRepository.init())
                .block()

        println("cows size = " + cowRepository.findAll()
                .collectList()
                .block()!!.size)
    }
}
